packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -722,14 +725,25 @@ export const RemoteServerFactory = {
                     })
 
                     // We pattern match and add client secrets only to endpoints that
-                    // match the regex specified by options.applySLASPrivateClientToEndpoints
+                    // match the regex specified by options.applySLASPrivateClientToEndpoints and are
+                    // not disallowed by doNotApplyPrivateClientSecretToEndpoints
                     // (see option defaults at the top of this file).
                     // Other SLAS endpoints, ie. SLAS authenticate (/oauth2/login) and
                     // SLAS logout (/oauth2/logout), use the Authorization header for a different
                     // purpose so we don't want to overwrite the header for those calls.
-                    if (incomingRequest.path?.match(options.applySLASPrivateClientToEndpoints)) {
+                    if (
```

Comment:
1. can we split the if statements to make it more readable?
2. should we be a bit more disruptive to throw error or warnings? since this is security related

```
if (isTrusted) {
  throw new Error('It is not allowed to include /trusted-system endpoints in `applySLASPrivateClientToEndpoints`')
}

if (incomingRequest.path?.match(options.applySLASPrivateClientToEndpoints)) {
  // rest of the code
}
```

Reply:
I've adjusted the proxy to do the following:

* Throw an error if the applySLASPrivateClientToEndpoints regex will match the SLAS trusted-system endpoint
* Throw a 403 if a request to the trusted-system endpoint is received